network_simulator/flow.py

```diff
@@ -4,8 +4,11 @@
 """
 
 from abc import ABCMeta, abstractmethod
+import copy
 from Queue import PriorityQueue
 
+import numpy as np
```

Comment:
The order is:

standard Python imports (alphabetized)

third-party imports (alphabetized); numpy is not installed by default

project-specific imports (alphabetized)